src/docs/guides/11-how-to-sanitize-user-generated-content-in-ftw/index.md

```diff
@@ -0,0 +1,71 @@
+---
+title: How to sanitize user-generated content in FTW
+slug: how-to-sanitize-user-generated-content-in-ftw
+updated: 2019-02-15
+category: guides
+ingress:
+  This guide describes how to sanitize user-generated content on Flex
+  Template for Web (FTW).
+published: true
+---
+
+User-generated content (UGC) is a big source of Cross-Site Scripting
+(XSS) vulnerabilities so one should pay attention to it.
+
+By default,
+[React DOM escapes any values embedded in JSX](https://reactjs.org/docs/introducing-jsx.html#jsx-prevents-injection-attacks)
+before rendering them. This makes customization a bit more
+straightforward from XSS perspective.
+
+However, there are other XSS vulnerabilities to consider. User-generated
+content should be validated if passed to component props. For example,
+`href` attribute in `<a>` tag could potentially contain XSS attack
+vector since it allows JavaScript execution like
+`href="javascript:alert('An XSS vulnerability found.')"`.
+
+## First things first: set up Content Security Policy
+
+One of the most advanced safety measures in modern browsers is Content
+Security Policy (CSP). When it is used, it helps to detect and mitigate
+certain types of attacks, including XSS and data injection attacks. Read
+more from our documentation on
+[how to set up Content Security Policy](/guides/content-security-policy/)
+
+## Sanitize user-generated content when updating entities
+
+We have added an example of how user-generated content could be
+sanitized when entities are queried and updated to Redux store. Those
+examples can be found from the utility file
+[src/util/sanitize.js](https://github.com/sharetribe/flex-template-web/blob/master/src/util/sanitize.js).
+
+You should modify those sanitize functions (e.g. `sanitizeUser`) to
+include any extendedData you have created for your marketplace. This is
+important if the data is used as a props/attribute in components (e.g.
+`<div attr=...ESCAPE UNTRUSTED DATA BEFORE PUTTING HERE...>content`). As
```

Comment:
I guess this depends on the attribute, right? So something like `<a data-user-href={user.publicData.link}>` is fine but `<a href={user.publicData.link}>` is not.

So React ensures that you cannot do something like `user.publicData.link='#" onClick="javascript:alert(\'oh no\')" hacked="true'` which would end up as `<a data-user-href="#" onClick="javascript:alert(\'oh no\')" hacked="true">` in a simple template language that only does string concatenation and not actual DOM operations.

Am I correct? This is mostly about understanding the issue, I don't think we even need to mention that some attributes are safe...

Reply:
Yes. 
Your example would convert to something like
```
<a data-user-href="#" onClick="javascript:alert('oh no')" hacked="true">
```